### PR TITLE
Simplified computation of smoothed cross entropy loss in CrossEntropyMetric

### PR DESCRIPTION
**Pull request by fhieber**
found a way to compute smoothed cross entropy without explicitly representing the smoothed distribution. This PR uses this alternative way in the CrossEntropyMetric, which is for now only relevant for the Eve optimizer.

All existing tests still pass.

#### Pull Request Checklist ##
- [x] Changes are complete (if posting work-in-progress code, prefix your pull request title with '[WIP]'
until you can check this box.
- [x] Unit tests pass (`pytest`)
- [x] System tests pass (`pytest test/system`)
- [x] Passed code style checking (`./style-check.sh`)
- [x] You have considered writing a test

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: sockeye/loss.py [hunk 1]
```diff
@@ -145,40 +145,48 @@ def __init__(self,
...
 
     def update(self, labels, preds):
         for label, pred in zip(labels, preds):
             batch_size = label.shape[0]
             label = label.as_in_context(pred.context).reshape((label.size,))
-            # Ignore padding
-            # TODO: contribute ignoring padding for cross-entropy back to MXNet
-            ignore = (label == C.PAD_ID).astype(dtype=pred.dtype)
 
+            logprob = mx.nd.log(pred + 1e-8)
```
Comment (tdomhan) [inline sockeye/loss.py]: would it make sense to do this with `mx.sym.clip`?

Comment (fhieber) [reply sockeye/loss.py]: Ok, will use the same pattern as in MXNET's perplexity metric: https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/metric.py#L749

**Comment by fhieber** [comment]
ping @tdomhan
